packages/react-components/lib/error-overlay.tsx

```diff
@@ -0,0 +1,76 @@
+import React from 'react';
+import { makeStyles } from '@material-ui/core';
+import ErrorIcon from '@material-ui/icons/Error';
+import { Typography, Grid } from '@material-ui/core';
+
+const useStyles = makeStyles((theme) => ({
+  errorIcon: {
+    color: theme.palette.error.main,
+  },
+  errorMsg: {
+    margin: '0.5rem',
+  },
+  errorDisabled: {
+    pointerEvents: 'none',
+    filter: 'blur(.25rem)',
+    gridArea: '1 / 1',
+    opacity: 0.6,
+  },
+  overlay: {
+    gridArea: '1 / 1',
+    marginTop: '1rem',
+    backdropFilter: 'blur(.5rem)',
+    padding: theme.spacing(1),
+  },
+  container: {
+    display: 'grid',
+  },
+  disableSelect: {
+    userSelect: 'none',
+  },
+}));
+
+export interface ErrorOverlayProps {
+  errorMsg?: string;
+  children: JSX.Element | null;
+  overrideErrorStyle?: string;
+}
+
+export const ErrorOverlay = (props: ErrorOverlayProps): JSX.Element => {
```

Comment:
Can you check if this component is re-rendering non-stop on the dashboard, with the chrome performance tool or the React profiler? 

If it is we can try adding a `React.memo`